Firmware/MotorControl/motor.cpp

```diff
@@ -186,10 +162,15 @@ float Motor::effective_current_lim() {
     } else {
         current_lim = std::min(current_lim, axis_->motor_.current_control_.max_allowed_current);
     }
-    // Thermal limit
-    current_lim = std::min(current_lim, thermal_current_lim_);
 
-    return current_lim;
+    // Apply axis current limiters
+    for (const CurrentLimiter* const limiter : axis_->current_limiters_) {
+        current_lim = std::min(current_lim, limiter->get_current_limit(config_.current_lim));
```

Comment:
The current limiter doesn't "know" the configured current limit. It's purpose of the argument is not only being a default but also what concrete current limit to return. An alternative would be for current limiters to not return a concrete current value but rather a current scale between 0.0 and 1.0. And then the logic would simply be find the smallest scaler and apply it to the configured current limit.

An optional would be overkill I think because we would need the `min` call anyway. Suppose we have two current limiters with a configured limit of 20A.

1. First current limiter says current limit is 15A.
2. Second current limiter says current limit is 12A.

Then we would still need to find the minimum of all the current limits to know which to actually apply. Introducing an optional is simply another case without (to me) clear benefit. 

Anyway the best way forward I think would be to rewrite this to a scaler based implementaiton. `get_current_limit` would not have any arguments and return a float between 0.0 and 1.0. The `std::min` would remain and the `effective_current_lim_` will become `current_lim * current_lim_scaler`.

Reply:
Ah ok, the value is also used as a base current limit to scale from, I see.
Now that I understand what is going on, I think it is good as is. Changing it to using `current_lim_scaler` will preclude other limiting scenarios that are an absolute limit rather than ratiometric de-rating.

Perhaps we can rename it to `base_current_lim` instead of user limit, it may make it a bit clearer that the main purpose is to do scaling off, rather than a user default (which is what I misinterpreted it as).

Reply:
Fixed.